### PR TITLE
Saving and loading weights

### PR DESCRIPTION
**Pull request by jan-nicklaus**
While using sentence transformers, I noticed the following "problem": When using colab, it is advisable to chunk your dataset into multiple parts to train on, creating a loop of model.fit(), saving and loading. But when calling model.fit() for the second time, the weights are appearantly dropped. This can be fixed by saving and loading the torch state dict for model and optimizers, which is what I implemented here. I also updated the section on continuing training in the docs. I believe this small addition of a few lines of code could help out users, especially if training on colab hardware.

### CODE REVIEW

**Comment by tomaarsen** [comment]
Hello!

Apologies for the delay! Resuming training has been integrated in the v3.0 update: [`SentenceTransformerTrainer.train`](https://sbert.net/docs/package_reference/sentence_transformer/trainer.html#sentence_transformers.trainer.SentenceTransformerTrainer.train) accepts a `resume_from_checkpoint` parameter where you can pass a path to a training checkpoint with the trainer state. You can now resume without any loss of training progress.

- Tom Aarsen
